atc/lidar/scanner.go

```diff
@@ -58,13 +69,30 @@ func (s *scanner) Run(ctx context.Context) error {
 			}()
 			defer waitGroup.Done()
 
-			s.check(spanCtx, resource, resourceTypes, resourceTypesChecked)
+			s.check(ctx, resource, resourceTypes, resourceTypesChecked)
 		}(resource, resourceTypes)
 	}
+}
 
-	waitGroup.Wait()
+func (s *scanner) scanUncheckedResourceTypes(ctx context.Context, resourceTypes db.ResourceTypes, waitGroup *sync.WaitGroup, resourceTypesChecked *sync.Map) {
+	logger := lagerctx.FromContext(ctx)
+	for _, resourceType := range resourceTypes {
+		waitGroup.Add(1)
 
-	return nil
+		go func(resourceType db.ResourceType, resourceTypes db.ResourceTypes) {
```

Comment:
Minor thing, but we don't need to create a goroutine when the resource type is already checked. e.g. this could be:

```go
for _, resourceType := range resourceTypes {
	if _, exists := resourceTypesChecked.LoadOrStore(resourceType.ID(), true); exists {
		// check has already been created for resource type
		continue
	}
	waitGroup.Add(1)

	go func(resourceType db.ResourceType, resourceTypes db.ResourceTypes) {
		defer func() {
			err := util.DumpPanic(recover(), "scanning resource type %d", resourceType.ID())
			if err != nil {
				logger.Error("panic-in-scanner-run", err)
			}
		}()
		defer waitGroup.Done()

		s.check(ctx, resource, resourceTypes, resourceTypesChecked)
	}(resourceType, resourceTypes)
}
```